### PR TITLE
Configurable reward and penalty scores

### PR DESCRIPTION
**Pull request by jaudick**
Right now the goal reward can be a float but before this change it was an int. How do we feel about it being a float if configured that way?

### CODE REVIEW

File: docs/source/install.rst [hunk 1]
```diff
@@ -121,6 +121,14 @@ To use a specific configuration, you can either pass in a file path or dictionar
 Config File Properties
 **********************
 
+goal_reward
+^^^^^^^^^^^^^^^
+
+(float, optional)
+
+Changes the postive reward recieved for achieving a goal.
```
Comment (ThomasSchellenbergNextCentury) [inline docs/source/install.rst:129]: Would be good for these three options to list their respective defaults.

File: tests/test_reward.py [hunk 2]
```diff
@@ -565,6 +599,46 @@ def test_transferral_reward_on_top_of_with_pickedup_object(self):
...
+            # create upper plane (y = 1) + i
+            obj['objectBounds']['objectBoundsCorners'].append(
+                {'x': 0.0, 'y': 1.0 + i, 'z': 0.0})
+            obj['objectBounds']['objectBoundsCorners'].append(
+                {'x': 1.0, 'y': 1.0 + i, 'z': 0.0})
+            obj['objectBounds']['objectBoundsCorners'].append(
+                {'x': 1.0, 'y': 1.0 + i, 'z': 1.0})
+            obj['objectBounds']['objectBoundsCorners'].append(
+                {'x': 0.0, 'y': 1.0 + i, 'z': 1.0})
+            obj['position'] = {'x': 0.5, 'y': 0.0 + i, 'z': 0.5}
```
Comment (ThomasSchellenbergNextCentury) [inline tests/test_reward.py:632]: FYI In the future you may find it easier to do something like this:

```python
obj = {
    "objectId": str(i),
    "objectBounds": {
        "objectBoundsCorners": [
            {'x': 0.0, 'y': 0.0 + i, 'z': 0.0},
            {'x': 0.0, 'y': 0.0 + i, 'z': 0.0},
            {'x': 1.0, 'y': 0.0 + i, 'z': 0.0},
            {'x': 1.0, 'y': 0.0 + i, 'z': 1.0},
            {'x': 0.0, 'y': 0.0 + i, 'z': 1.0},
            {'x': 0.0, 'y': 1.0 + i, 'z': 0.0},
            {'x': 1.0, 'y': 1.0 + i, 'z': 0.0},
            {'x': 1.0, 'y': 1.0 + i, 'z': 1.0},
            {'x': 0.0, 'y': 1.0 + i, 'z': 1.0}
        ]
    },
    "position": {'x': 0.5, 'y': 0.0 + i, 'z': 0.5}
}
```

**Review by ThomasSchellenbergNextCentury — CHANGES_REQUESTED** [review]
I'm getting a bunch of unit test errors. They're all `TypeError: float() argument must be a string or a number, not 'NoneType'`. Is it just me?

**Comment by jaudick** [comment]
> I'm getting a bunch of unit test errors. They're all `TypeError: float() argument must be a string or a number, not 'NoneType'`. Is it just me?

Not just you, looks like I missed that configs were being set in this test file. I had only tested the reward unit tests. I will get on this

**Comment by jaudick** [comment]
@ThomasSchellenbergNextCentury the error is fixed

**Review by deanwetherby — COMMENTED** [review]
Some unit tests specifically for the score adjustment function would be good to add to this PR. There are numerous conditions to test for including penalty values of None and if the task has already been completed successfully.

**Comment by jaudick** [comment]
> There are numerous conditions to test for including penalty values of None
@deanwetherby This is confusing me a little. Are you saying that None would be passed in intentionally to have penalty values of zero?

> and if the task has already been completed successfully.
Also confused here. Are you asking to test multiple values of goal_reward configs like: test 500, test 1.5, test 0.1? or test if penalties happen after the goal is achieved?

**Comment by deanwetherby** [comment]
> @deanwetherby This is confusing me a little. Are you saying that None would be passed in intentionally to have penalty values of zero?
> 
> Are you asking to test multiple values of goal_reward configs like: test 500, test 1.5, test 0.1? or test if penalties happen after the goal is achieved?

You'll want to check each branching condition in the function. What do you expect to happen if step_penalty or lava_penalty is None? What if the current_score is 1 vs not 1? etc. The adjusted score output is going to vary on those conditions.

Unit tests are also a form of documentation so another developer can see how the function gets used and what to expect for an output.

```
        step_penalty = STEP_PENALTY if step_penalty is None else step_penalty
        lava_penalty = LAVA_PENALTY if lava_penalty is None else lava_penalty
        if current_score == 1:
            return current_score - ((number_steps - 1) * float(step_penalty))

        if steps_on_lava is None:
            steps_on_lava = 0
```

**Comment by jaudick** [comment]
@deanwetherby I added three additional unit tests based on your suggestions

**Review by deanwetherby — APPROVED** [review]

**Review by ThomasSchellenbergNextCentury — APPROVED** [review]

**Review by brianpippin — APPROVED** [review]

**Pull request merged**